File: src/lib/types/playlist.ts
export interface Playlist {
  readonly id: string;
  title: string;
  description: string | undefined;
  thumbnail_path: string | undefined;
  // TODO: rules
  rules: string;
  sort_order: number;
}

export interface PlaylistFolder {
  readonly id: string;
  title: string;
  description: string | undefined;
  sort_order: number;
}

File: src-tauri/improvie-logic/src/model/playlist.rs
use serde::{Deserialize, Serialize};

use crate::{Uuid, logic::rule::Rule};

#[derive(Clone, Serialize, Deserialize)]
pub struct PlaylistFolder {
    pub id: Uuid,
    pub title: String,
    pub description: Option<String>,
    pub sort_order: u32,
}

#[derive(Clone, Serialize, Deserialize)]
pub struct Playlist {
    pub id: Uuid,
    pub title: String,
    pub description: Option<String>,
    pub thumbnail_path: Option<String>,
    pub rules: Vec<Rule>,
    pub sort_order: u32,
}
